Assets/Shopify/UIToolkit/Components/RemoteImageLoader.cs

```diff
@@ -0,0 +1,76 @@
+namespace Shopify.UIToolkit {
+    using UnityEngine;
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    public interface ImageLoader {
+        string GetError();
+
+        Texture2D GetTexture();
+
+        IEnumerator Load(string url);
+    }
+
+    public class UnityImageLoader : ImageLoader {
```

Comment:
this feels kinda Yagni... when are we going to need a non-unityloader in the unity UI Kit?

Reply:
oh i see in the tests... clever!

Reply:
Yup - mostly just to stub out the network with a mocked call.